moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -1989,53 +1990,19 @@ double moveit::core::RobotState::computeCartesianPath(const JointModelGroup* gro
   return percentage_solved;
 }
 
-namespace
-{
-static inline void updateAABB(const Eigen::Affine3d& t, const Eigen::Vector3d& e, std::vector<double>& aabb)
-{
-  Eigen::Vector3d v = e / 2.0;
-  Eigen::Vector3d c2 = t * v;
-  v = -v;
-  Eigen::Vector3d c1 = t * v;
-  if (aabb.empty())
-  {
-    aabb.resize(6);
-    aabb[0] = c1.x();
-    aabb[2] = c1.y();
-    aabb[4] = c1.z();
-    aabb[1] = c2.x();
-    aabb[3] = c2.y();
-    aabb[5] = c2.z();
-  }
-  else
-  {
-    if (aabb[0] > c1.x())
-      aabb[0] = c1.x();
-    if (aabb[2] > c1.y())
-      aabb[2] = c1.y();
-    if (aabb[4] > c1.z())
-      aabb[4] = c1.z();
-    if (aabb[1] < c2.x())
-      aabb[1] = c2.x();
-    if (aabb[3] < c2.y())
-      aabb[3] = c2.y();
-    if (aabb[5] < c2.z())
-      aabb[5] = c2.z();
-  }
-}
-}
-
 void robot_state::RobotState::computeAABB(std::vector<double>& aabb) const
 {
   BOOST_VERIFY(checkLinkTransforms());
 
   aabb.clear();
+  core::AABB _aabb;
   std::vector<const LinkModel*> links = robot_model_->getLinkModelsWithCollisionGeometry();
   for (std::size_t i = 0; i < links.size(); ++i)
   {
-    const Eigen::Affine3d& t = getGlobalLinkTransform(links[i]);
+    Eigen::Affine3d t = getGlobalLinkTransform(links[i]);  // intentional copy, we will translate
```

Comment:
can we use better variable names here besides one letter names? i know this was already here like this

Reply:
Glad! I just wanted the PR to be minimal, but if you think it's time to become more verbose, I'm in ;)

Reply:
you have a good point, you can do it in a separate PR if you want

Reply:
feel free to change it as part of this request too, if you like.